### ISSUE TITLE
Rename `serverSupabaseServiceRole` to `getSupabaseClient`

### ISSUE THREAD

**Issue by pi0** [nuxt-modules/supabase#67]
Hi! Since new h3 and nitro utils are using `get*(event, ...)` pattern, we were talking with @Atinux and I would suggest to deprecate `serverSupabaseServiceRole` server util to replace with `getSupabaseClient`.

**Comment by Atinux** [comment]
Indeed, what is the best way to keep the old naming and show a deprecation message to avoid a breaking change?

**Comment by larbish** [comment]
I think I'll keep both `serverSupabaseClient` and `getSupabaseClient` here: https://github.com/nuxt-modules/supabase/blob/main/src/module.ts#L146 for the moment. 

But what is the way to display a breaking change message ?

**Comment by pi0** [comment]
I would suggest to simply add a `/** @deprecated use new method */` for soft deprecation and drop in next major version of module whenever it came. See https://github.com/unjs/h3/pull/156
